src/main/java/sparksoniq/jsoniq/runtime/iterator/operational/InstanceOfIterator.java

```diff
@@ -52,12 +52,17 @@ public Item next() {
                 items.add(_child.next());
             _child.close();
             this._hasNext = false;
+
+            //... instance of ()
+            if (_sequenceType.isEmpty()) return ItemFactory.getInstance().createBooleanItem(false);
```

Comment:
Could we also rename isEmpty() to isEmptySequence() to make sure there is no confusion?

Reply:
There is a missing test for whether items is empty. If items is empty and _sequenceType.isEmpty() is true, then the boolean item true should be returned. If items is non-empty and _sequenceType.isEmpty() is true, then the boolean item false should be returned.

For some reason that I don't understand just with a quick look, it seems that something odd happens when the input sequence is empty at the moment; fixing this will make getResultIfEmpty() superfluous.

Reply:
I am not sure I understood correctly what test would be missing. I added all the possible combinations of:

- x instance of y
- x instance of ()
- () instance of y
- () instance of y?
- () instance of ()

with x != () and y != ().

Reply:
I think that it's all good. Only + and * were missing (see my other comment).

Reply:
From revision 3, i.e. commit 16e34f9, I added those tests as well.